File: LeetCode/1_TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // iterate through the given array
    for (var i = 0; i < nums.length; i++) {
        // iterate within to vet every number at i with every other number at j
        for (var j = i + 1; j < nums.length; j++) {
            if (nums[i] + nums[j] === target) {
                return [i, j];
            }
        }
    }

};


/* NOTES:


*/

File: LeetCode/1_TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        int[] solution = new int[2];
        List<int> taken = new List<int>();

        for (int i = 0; i < nums.Length - 1; i++)
        {
            int pointer = nums.Length - 1;

            if (taken.Contains(nums[i]))
            {
                continue;
            }

            taken.Add(nums[i]);

            while (pointer > i)
            {
                if (nums[i] + nums[pointer] == target)
                {
                    solution[0] = i;
                    solution[1] = pointer;
                    return solution;
                }
                pointer--;
            }
        }
        return solution;
    }
}